tests/release/releaseLib.go

```diff
@@ -303,18 +303,27 @@ func CreateOpaqueSecret(
 		secretData[field] = decodedValue
 	}
 
+	new_secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: secretData,
+	}
+
 	secret, err := fw.AsKubeAdmin.CommonController.GetSecret(namespace, secretName)
-	if secret == nil || errors.IsNotFound(err) {
-		secret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretName,
-				Namespace: namespace,
-			},
-			Type: corev1.SecretTypeOpaque,
-			Data: secretData,
-		}
+	if secret != nil {
+		_, err = fw.AsKubeAdmin.CommonController.UpdateSecret(namespace, new_secret)
+		Expect(err).ToNot(HaveOccurred())
+		return
+	}
 
-		_, err = fw.AsKubeAdmin.CommonController.CreateSecret(namespace, secret)
+	if errors.IsNotFound(err) {
+		_, err = fw.AsKubeAdmin.CommonController.CreateSecret(namespace, new_secret)
 		Expect(err).ToNot(HaveOccurred())
+		return
 	}
+
+	Expect(err).ToNot(HaveOccurred())
 }
```

Comment:
Hi @jinqi7 

the `tests/release/releaseLib.go` file was updated in this PR, but when I look at the ginkgo command that was run for this PR, the label filter was set to:
```
"--label-filter=!upgrade-create && !upgrade-verify && !upgrade-cleanup && !release-pipelines"
```
([full log](https://app-artifact-browser.apps.rosa.konflux-qe.zmr9.p3.openshiftapps.com/e2e-tests/konflux-e2e-dj9vw/e2e-tests.log))

i.e. it didn't run the `release-pipelines` tests that are affected by this change. Would it make sense to update the test selection rule for such case?

Reply:
Yes. It's the test selection rule in e2e repo. I submitted the PR https://github.com/konflux-ci/e2e-tests/pull/1556/files.